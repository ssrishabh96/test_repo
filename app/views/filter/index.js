import Filter from './filter.view';

export default Filter;
