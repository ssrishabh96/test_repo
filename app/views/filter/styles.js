import { StyleSheet, Platform } from 'react-native';
import colors from 'styles/colors';

export default StyleSheet.create({
  categoryColor: {
    width: '45%',
    backgroundColor: colors.GRAY_2,
  },
  activeCategory: {
    backgroundColor: '#a6a6a6',
  },
  categoryButton: {
    padding: 6,
    flexDirection: 'row',
    flex: 1,
    borderBottomWidth: 1,
    borderBottomColor: colors.GRAY_DARK_1,
  },
  categoryText: {
    flex: 1,
    color: colors.GRAY_DARK_1,
    fontWeight: 'bold',
    fontSize: 18,
    borderBottomColor: '#cccccc',
  },
  rowOptions: {
    flexDirection: 'row',
    padding: 10,
    borderBottomWidth: 1,
    alignItems: 'center',
  },
  checkBox: { paddingRight: 5 },
  valueWrap: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  valueText: {
    fontSize: 16,
    color: colors.GRAY_DARK_1,
    flex: 1,
  },
  valueCount: {
    fontSize: 16,
    color: colors.GRAY_DARK_1,
    textAlign: 'right',
  },
  buttonText: {
    color: colors.LIGHT_YELLOW,
    fontSize: 16,
    fontWeight: 'bold',
    marginLeft: 10,
  },
  title: {
    flexDirection: 'row',
    height: 60,
    justifyContent: 'space-between',
    backgroundColor: colors.COPART_BLUE,
  },
  headerView: {
    backgroundColor: colors.GRAY_DARK_1,
    flexDirection: 'row',
    justifyContent: 'space-between',
    height: 45,
    alignItems: 'center',
    paddingHorizontal: 5,
  },
  topButtons: {
    top: 30,
    padding: 2,
  },
  headerText: {
    padding: 2,
    color: colors.WHITE,
    fontWeight: 'bold',
    fontSize: 16,
    paddingBottom: 3,
  },
  badgeView: {
    paddingHorizontal: 8,
    paddingVertical: 3,
    backgroundColor: colors.LIGHT_YELLOW,
    borderRadius: 20,
    borderWidth: 1,
    borderColor: 'transparent',
  },
  navbar: {
    height: Platform.OS === 'ios' ? 65 : 60,
    paddingTop: Platform.OS === 'ios' ? 20 : 0,
    paddingHorizontal: 10,
    flexDirection: 'row',
    backgroundColor: colors.COPART_BLUE,
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  navbarButton: { color: 'white' },
  navbarTitle: {
    color: 'white',
    fontWeight: 'bold',
    fontSize: 17,
    ...(Platform.OS === 'android' ? { flex: 1, textAlign: 'left', paddingLeft: 20 } : {}),
  },
});
