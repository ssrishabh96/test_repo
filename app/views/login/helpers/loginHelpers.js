import { pathOr, compose, join, head, propEq, find } from 'ramda';

export const buildSession = (data) => {
  const now = Date.now();
  const tokens = {
    access: {
      type: data.token_type,
      value: data.access_token,
      expiresIn: data.expires_in,
      timestamp: now,
      expirationTimestamp: now + data.expires_in * 1000,
    },
    refresh: {
      type: data.token_type,
      value: data.refresh_token,
    },
  };
  const user = {
    id: data.entity_id,
    name: data.entity_name,
    country: data.entity_country,
    type: data.entity_type,
    isTempPass: data.temp_password_flag,
  };
  return {
    tokens,
    user,
  };
};

export const checkIfSingleVendorPresent = (res) => {
  if (res.vendors.length === 1) {
    const vendor = res.vendors[0];
    return {
      result: true,
      vendor,
    };
  }
  return { result: false, vendor: null };
};

export const getActiveVendorId = pathOr(-1, [
  'login',
  'user',
  'profiles',
  'activeProfile',
  'vendor',
  'vendor_id',
]);

export const getVendorByVendorId = (res, id) => find(propEq('vendor_id', id))(res.vendors);
