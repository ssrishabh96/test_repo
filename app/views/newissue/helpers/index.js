// @flow

export const getHeaderText = (entity: string, data: Object): string => {
  let headerText = '';
  if (entity && entity === 'trip') {
    if (data.length === 1) {
      headerText = `TRIP# ${data[0]}`;
    } else {
      headerText = `TRIP# ${data[0]} + ${data.length - 1} more`;
    }
  } else if (entity && entity === 'lot') {
    headerText = `LOT # ${data.number}`;
  }
  return headerText;
};
