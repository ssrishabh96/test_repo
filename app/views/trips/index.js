import Trips from './trips.view';

export default Trips;
