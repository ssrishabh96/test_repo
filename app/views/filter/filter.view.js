import React, { Component } from 'react';
import { View, Modal, StyleSheet } from 'react-native';
import { defaultNavStyles } from 'styles';
import colors from 'styles/colors';

import {
  getInitialField,
  getInitialValues,
  prepareFinalSelected,
  prepareInitialSelected,
  prepareData,
  getAvailableFilters,
  getSelectedCount,
} from './filter.helpers';

import FullScreenLoader from 'components/custom/FullScreenLoader';
import Header from './components/Header';
import CategoryList from './components/CategoryList';
import ValuesList from './components/ValuesList';

type Props = {
  +navigator: Object,
  +applyFilter: (selectedFilters: Object, totalCount: number) => any,
  +selectedFilters: {
    [string]: string[],
  },
  +totalCount: number,
  +bucket: string,
  +isVisible: boolean,
  +closeFilters: () => null,
};

export default class Filter extends Component<Props> {
  state = {
    loading: true,
    selectedFilters: {},
    totalCount: 0,
    field: '',
    values: [],
    availableFilters: [],
  };
  componentWillMount() {
    if (this.props.isVisible) this.onLaunch(); // ? ??
  }
  componentDidUpdate(prevProps) {
    if (!prevProps.isVisible && this.props.isVisible) this.onLaunch();
  }
  onLaunch = () => {
    this.setState({
      loading: true,
      selectedFilters: {},
      totalCount: 0,
      field: '',
      values: [],
      availableFilters: [],
    });
    getAvailableFilters(this.props.bucket)
      .then((data) => {
        const filters = prepareData(data);
        const selected = prepareInitialSelected(this.props.selectedFilters, data);
        this.setState({
          selectedFilters: selected,
          totalCount: getSelectedCount(selected),
          loading: false,
          field: getInitialField(filters),
          values: getInitialValues(filters),
          availableFilters: filters,
        });
      })
      .catch((error) => {
        // console.log error
        this.props.navigator.showInAppNotification({
          screen: 'CopartTransporter.ShowInAppNotification',
          passProps: {
            type: 'warning',
            content: 'Unable to retrieve filters',
          },
        });
        this.props.closeFilters();
      });
  };
  onItemSelected = (field: string, label: string, values: Array) => {
    this.setState({
      field,
      values,
    });
  };
  onCheckboxSelected = (field: string, code: string) => {
    this.setState({
      selectedFilters: {
        ...this.state.selectedFilters,
        [field]: this.state.selectedFilters[field]
          ? {
            ...this.state.selectedFilters[field],
            [code]: !this.state.selectedFilters[field][code],
            count: this.state.selectedFilters[field][code]
              ? this.state.selectedFilters[field].count - 1
              : this.state.selectedFilters[field].count + 1,
          }
          : { [code]: true, count: 1 },
      },
      totalCount:
        this.state.selectedFilters[field] && this.state.selectedFilters[field][code]
          ? this.state.totalCount - 1
          : this.state.totalCount + 1,
    });
  };
  getSelectedFilterCount = field =>
    this.state.selectedFilters[field] ? this.state.selectedFilters[field].count : 0;
  resetFilter = () => {
    this.setState({
      selectedFilters: {},
      totalCount: 0,
    });
  };
  handleOnApply = () => {
    const selected = prepareFinalSelected(this.state.selectedFilters);
    this.props.applyFilter(selected, this.state.totalCount);
    this.props.closeFilters();
  };
  render() {
    const { field, values, selectedFilters, totalCount, loading, availableFilters } = this.state;
    return (
      <Modal
        transparent
        visible={this.props.isVisible}
        animationType="slide"
      >
        <View style={styles.container}>
          <Header
            bucket={this.props.bucket}
            totalCount={totalCount}
            resetFilter={this.resetFilter}
            handleOnApply={this.handleOnApply}
            closeFilters={this.props.closeFilters}
          />
          {loading ? (
            <View style={styles.innerContainer}>
              <FullScreenLoader />
            </View>
          ) : (
            <View style={styles.innerContainer}>
              <CategoryList
                data={availableFilters}
                field={field}
                onPress={this.onItemSelected}
                selectedFilters={selectedFilters}
              />
              <ValuesList
                data={values}
                field={field}
                onPress={this.onCheckboxSelected}
                selectedFilters={selectedFilters}
              />
            </View>
          )}
        </View>
      </Modal>
    );
  }
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'rgba(10, 8, 10, 0.78)',
    // marginTop: 64,
  },
  innerContainer: { flex: 1, flexDirection: 'row', backgroundColor: 'white' },
});
