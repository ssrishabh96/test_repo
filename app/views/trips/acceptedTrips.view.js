import React, { Component } from 'react';
import { View } from 'react-native';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import renderIf from 'render-if';
import { find, propEq, isEmpty } from 'ramda';

import { getUserRole } from 'views/login/login.redux';
import {
  getAcknowledgedTrips,
  setMultiselectMode,
  resetMultiselectMode,
  toggleSelection,
  distributeTrips,
  showSearch,
  resetSearch,
  resetAllFilters,
  setSearchQuery,
  setSelectedFilters,
  activateRefresh,
  handleSortChangeForLotList,
  toggleModal,
  toggleFilterVisibility,
  hideSearch,
} from './trips.actions';

import FullScreenLoader from 'components/custom/FullScreenLoader';
import TripsListHeader from './components/ViewHeader';
import TripList from './components/TripList';
import SearchBar from 'components/custom/SearchBar';
import SortModal from 'components/custom/SortModal';
import FilterModal from 'views/filter/filter.view';
import { DistributeButtons } from '../lotlist/components/FAB';
import AssignedTo from 'views/distributed/components/AssignedTo';

import { DRIVER as DRIVER_ROLE } from 'constants/user/roles';
import { getNavBarRightButtons } from 'config/UserManager';
import NavListener from 'utils/NavigationListener';
import Locale from 'utils/locale';
import { countLots, hasDistributableTrips } from './helpers/tripHelpers';
import { defaultNavStyles } from 'styles';

// TODO: may need a pull to refresh for assigned trips
type Props = {
  +navigator: Object,
  +resetMultiselectMode: () => any,
  +setMultiselectMode: () => any,
  +toggleSelection: () => any,
  +selectedFilters: {
    [string]: string[],
  },
  +sort: {
    isModalVisible: boolean,
    selectedField: Object,
  },
  +role: number,
  +setSelectedFilters: () => any,
  +toggleFilterVisibility: () => any,
  +getAcknowledgedTrips: () => any,
  +resetAllFilters: () => any,
  +filterCount: number,
  +filterIsVisible: boolean,
  +isLoading: boolean,
  +isLoadingNextPage: boolean,
  +multiselect: Object,
  +selected: Object,
  +handleSortChangeForLotList: () => any,
  +toggleModal: () => any,
  +hideSearch: () => any,
  +page: number,
  +totalCount: number,
  +lotTotalCount: number,
};
class AcceptedTrips extends Component<Props> {
  constructor(props) {
    super(props);
    const { navigator } = this.props;
    navigator.setOnNavigatorEvent(this.onNavigatorEvent.bind(this));
  }

  // componentWillMount() {
  //   this.props.navigator.setButtons({
  //     rightButtons: getNavBarRightButtons(hasDistributableTrips(this.props.trips), this.props.trips),
  //     animated: true,
  //   });
  // }

  onNavigatorEvent(event) {
    const { multiselect, trips, navigator, search } = this.props;
    if (event.type === 'DeepLink') {
      const links = event.link.split('/');
      if (links && links[0] === 'tab' && links[1] === '2' && links[2] === 'popToRoot') {
        navigator.popToRoot({
          animated: false,
        });
        return;
      }
      if (links[0] === 'acknowledged') {
        navigator.switchToTab({ tabIndex: 2 });
        NavListener.handleTabChange(2);
        const linkTripId = parseInt(links[1], 10);
        if (linkTripId) {
          const trip = find(propEq('tripId', linkTripId))(trips);
          if (trip && trip.lots) {
            const isAssignedAsDriver = trip.vendor_personnel_id !== null;
            const navbarRightButtons = getNavBarRightButtons(isAssignedAsDriver, trips);
            this.props.navigator.push({
              screen: 'CopartTransporter.LotList',
              title: Locale.translate('tab.Acknowledged.AcknowledgeLotsTitle'),
              passProps: {
                tripId: trip.tripId,
                tripName: trip.tripName,
                tripStatus: 'accepted',
                type: 'acceptedTrips',
                item: trip,
              },
              animated: false,
              navigatorButtons: {
                rightButtons: navbarRightButtons,
              },
              navigatorStyle: defaultNavStyles,
            });
          } else {
            // TODO: Dispatch proper error action
            throw new Error('Trip not found from deeplink');
          }
        }
      }
      return;
    }
    switch (event.id) {
      case 'backPress':
        navigator.switchToTab({ tabIndex: 0 });
        NavListener.handleTabChange(0);
        break;
      case 'willAppear':
        this.props
          .getAcknowledgedTrips({ navigator: this.props.navigator })
          .then(this.setNavigatorButtons);
        break;
      case 'distribute':
        if (multiselect.active) {
          this.props.resetMultiselectMode();
        } else {
          this.props.setMultiselectMode('distribute');
        }
        break;
      case 'didDisappear':
        if (multiselect.active) {
          this.props.resetMultiselectMode();
        }
        this.props.resetAllFilters('accepted', this.props.navigator);
        break;
      case 'search':
        if (!search.visible) {
          this.props.showSearch('accepted');
        }
        break;
      case 'bottomTabSelected':
        NavListener.handleTabChange(2);
        break;
      default:
        console.log('unhandled event', event.id); // eslint-disable-line
    }
  }
  setNavigatorButtons = () => {
    this.props.navigator.setButtons({
      rightButtons: getNavBarRightButtons(
        !hasDistributableTrips(this.props.trips),
        this.props.trips,
      ),
      animated: true,
    });
  };
  handleOnSelect = (tripId, item) => {
    const { multiselect, toggleSelection: tSelection, trips } = this.props;
    if (multiselect.active) {
      tSelection(tripId, item.dispatch_group_id);
    } else {
      const isAssignedAsDriver = item.vendor_personnel_id !== null;
      const navbarRightButtons = getNavBarRightButtons(isAssignedAsDriver, trips);
      this.props.navigator.push({
        screen: 'CopartTransporter.LotList',
        title: Locale.translate('tab.Acknowledged.AcknowledgeLotsTitle'),
        passProps: {
          tripId: item.tripId,
          tripName: item.tripName,
          tripStatus: 'accepted',
          type: 'acceptedTrips',
          item,
          comingFromTripList: true,
        },
        navigatorButtons: {
          rightButtons: navbarRightButtons,
        },
        navigatorStyle: defaultNavStyles,
      });
    }
  };

  handleOnAssign = (type: string) => {
    const { selected, trips, multiselect: { group: driverGroupId } } = this.props;
    const selectedTripCount = Object.values(selected).length;
    if (selectedTripCount === 0) {
      this.props.navigator.showInAppNotification({
        screen: 'CopartTransporter.ShowInAppNotification',
        passProps: {
          type: 'warning',
          content: 'Please select 1 or more Lots',
        },
        autoDismissTimerSec: 1.0,
      });
    } else {
      this.props.navigator.showModal({
        screen: 'CopartTransporter.AssignToList',
        title: 'Assign To',
        navigatorStyle: {},
        passProps: {
          mode: 'trips',
          type,
          selected,
          trips,
          driverGroupId,
          tripType: 'accepted',
        },
      });
    }
  };
  applyFilter = (selectedFilters, totalCount) => {
    this.props.setSelectedFilters(selectedFilters, totalCount, 'accepted', this.props.navigator);
    this.props.navigator.dismissModal(); // TODO fix this
  };
  handleFiltersVisible = () => {
    this.props.toggleFilterVisibility('accepted');
  };

  handleSearchQuery = (query: string) => {
    this.props.setSearchQuery(query, 'accepted', this.props.navigator);
  };

  handleSelectedItem = (field: number | string) => {
    this.props.handleSortChangeForLotList(field, 'accepted', this.props.navigator);
  };

  handleModalVisible = () => {
    this.props.toggleModal('accepted');
  };
  handleRefresh = () => {
    this.props
      .getAcknowledgedTrips({ navigator: this.props.navigator })
      .then(this.setNavigatorButtons);
  };
  handleLoadNextPage = () => {
    if (!this.props.isLoadingNextPage) {
      const { page } = this.props;
      const nextPage = page + 1;
      this.props.getAcknowledgedTrips({ navigator: this.props.navigator, page: nextPage });
    }
  };

  resetSearch = (bucket: string) => {
    const { search, navigator } = this.props;
    if (isEmpty(search.query)) {
      this.props.hideSearch(bucket);
    } else this.props.resetSearch(bucket, navigator);
  };

  render() {
    const {
      isLoading,
      isLoadingNextPage,
      trips,
      multiselect,
      selected,
      search,
      filterCount,
      sort,
      role,
    } = this.props;
    const renderDistributeButttonIfSplitMode = renderIf(
      multiselect.active && multiselect.type === 'distribute',
    );
    const count = trips.length;
    const renderIfHasTripsAndSearchVisible = renderIf(
      (search.visible && count > 0) || search.query,
    );

    const renderIfHasTripsOrFilterAndSearchNotVisible = renderIf(
      !search.visible && (count > 0 || filterCount !== 0),
    );

    if (isLoading) {
      return <FullScreenLoader />;
    }
    return (
      <View style={{ flex: 1 }}>
        {renderIfHasTripsAndSearchVisible(
          <SearchBar
            query={search.query}
            placeholder="Enter Trip Id"
            resetSearchBar={() => this.resetSearch('accepted')}
            onSubmit={this.handleSearchQuery}
          />,
        )}
        {renderIfHasTripsOrFilterAndSearchNotVisible(
          <TripsListHeader
            tripcount={this.props.totalCount}
            lotcount={this.props.lotTotalCount}
            filterCount={filterCount}
            selected={selected}
            multiselect={multiselect}
            showMultiselect={false} // remove the multiselect button
            disabled={trips.length === 0}
            toggleAckMode={this.handleToggleAckMode}
            showFilters={this.handleFiltersVisible}
            onCancel={this.props.resetMultiselectMode}
            toggleModal={this.handleModalVisible}
          />,
        )}
        <TripList
          listType={'acceptedTrips'}
          data={trips}
          isLoadingNextPage={isLoadingNextPage}
          handleOnPress={this.handleOnSelect}
          multiselect={multiselect}
          action={multiselect.type}
          selected={selected}
          onRefresh={this.handleRefresh}
          onEndReached={this.handleLoadNextPage}
          renderExtraView={
            role !== DRIVER_ROLE
              ? (
                disabled,
                isAssignedAsDriver,
                hasLotWithIssue,
                itemNotDistributable,
                groupDoesNotMatchSelected,
              ) => item => (
                <AssignedTo
                  name={item.responsible_party_name}
                  status={item.status}
                  type={'trip'}
                  multiselect={multiselect.active}
                  action={multiselect.type}
                  isAssignedAsDriver={isAssignedAsDriver}
                  hasLotWithIssue={hasLotWithIssue}
                  disabled={disabled}
                  itemNotDistributable={itemNotDistributable}
                  groupDoesNotMatchSelected={groupDoesNotMatchSelected}
                />
              )
              : null
          }
        />
        <SortModal
          type={'trip'}
          isModalVisible={sort.isModalVisible}
          selectedItem={this.handleSelectedItem}
          cancelSort={this.handleModalVisible}
          selectedField={sort.selectedField}
        />
        <FilterModal
          applyFilter={this.applyFilter}
          bucket="accepted"
          selectedFilters={this.props.selectedFilters}
          totalCount={this.props.filterCount}
          closeFilters={this.handleFiltersVisible} // TODO: this
          isVisible={this.props.filterIsVisible} // TODO: this
          navigator={this.props.navigator}
        />
        {renderDistributeButttonIfSplitMode(
          <DistributeButtons
            role={this.props.role}
            onItemClick={this.handleOnAssign}
          />,
        )}
      </View>
    );
  }
}

AcceptedTrips.defaultProps = {
  multiselect: { active: false, type: '' },
  selected: {},
  trips: [],
};

AcceptedTrips.propTypes = {
  trips: PropTypes.array, // eslint-disable-line
  // type: PropTypes.string.isRequired,
  search: PropTypes.object.isRequired, // eslint-disable-line
  setSearchQuery: PropTypes.func.isRequired,
  // toggleSearchVisible: PropTypes.func.isRequired,
  // activateRefresh: PropTypes.func.isRequired,
  resetSearch: PropTypes.func.isRequired,
  showSearch: PropTypes.func.isRequired,
};

const mapDispatchToProps = {
  getAcknowledgedTrips,
  setMultiselectMode,
  resetMultiselectMode,
  toggleSelection,
  distributeTrips,
  showSearch,
  resetSearch,
  resetAllFilters,
  setSearchQuery,
  setSelectedFilters,
  activateRefresh,
  handleSortChangeForLotList,
  toggleModal,
  toggleFilterVisibility,
  hideSearch,
};

const acceptedTripsSelector = state => ({
  trips: state.trips.accepted.trips,
  isLoading: state.trips.accepted.isLoading,
  isLoadingNextPage: state.trips.accepted.isLoadingNextPage,
  multiselect: state.trips.multiselect,
  selected: state.trips.selected,
  search: state.trips.accepted.search,
  selectedFilters: state.trips.accepted.filters.selectedFilters,
  filterCount: state.trips.accepted.filters.totalCount,
  filterIsVisible: state.trips.accepted.filters.isVisible,
  sort: state.trips.accepted.sort,
  role: getUserRole(state),
  totalCount: state.trips.accepted.count,
  page: state.trips.accepted.page,
  lotTotalCount: state.home.counts.acceptedTrips,
});

export default connect(acceptedTripsSelector, mapDispatchToProps)(AcceptedTrips);
