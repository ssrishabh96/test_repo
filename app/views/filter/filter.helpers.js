import {
  compose,
  head,
  prop,
  map,
  without,
  keys,
  values,
  reduce,
  reject,
  isEmpty,
  filter,
  equals,
  identity,
  forEachObjIndexed,
  mapObjIndexed,
  add,
} from 'ramda';
import moment from 'moment';
import { renameKeys } from 'utils/commonUtils';
import { bucketToStatusMap } from 'constants/Lot';
import * as tripsService from 'views/trips/trips.service';

export const getInitialField = compose(prop('field'), head);
export const getInitialValues = compose(prop('values'), head);

export const prepareInitialSelected = (selected, available) =>
  mapObjIndexed((i, k) =>
    reduce(
      (acc, val) => {
        if (available[k].indexOf(val) !== -1) {
          acc[val] = true;
          acc.count += 1;
        }
        return acc;
      },
      { count: 0 },
    )(i),
  )(selected);

export const prepareFinalSelected = compose(
  reject(isEmpty),
  map(compose(compose(without(['count']), keys), filter(equals(true)))),
);

const formatDate = date => moment(date).format('MM/DD/YYYY');
const formatPayment = payment => payment || 'Unknown';

export const categoryMap = {
  location_city: { label: 'City', getKeys: identity },
  location_zip: { label: 'Zip', getKeys: identity },
  trip_type_code: { label: 'Trip Type', getKeys: identity },
  vendor_name: { label: 'Tow Provider', getKeys: identity },
  is_residence: { label: 'Residence', getKeys: identity },
  is_towable: { label: 'Towable', getKeys: identity },
  loss_type_code: { label: 'Loss Type', getKeys: identity },
  lot_make: { label: 'Make', getKeys: identity },
  promised_date: { label: 'Promised Date', getKeys: formatDate },
  is_scheduled: { label: 'Scheduled', getKeys: identity },
  lot_stage_desc: { label: 'Lot Stage', getKeys: identity }, // ?
  tow_type_code: { label: 'Tow Type', getKeys: identity },
  active_issue_flag: { label: 'Has Active Issue', getKeys: identity },
  is_due_today: { label: 'Due Today', getKeys: identity },
  payment_mode: { label: 'Payment Mode', getKeys: formatPayment },
  lot_model_year: { label: 'Model Year', getKeys: identity },
  location_name: { label: 'Location Name', getKeys: identity },
  sublot_name: { label: 'Sublot Name', getKeys: identity },
  damage_type_code: { label: 'Damage Type', getKeys: identity }, // ?
};

const keymap = {
  residence_flag: 'is_residence',
  towable_flag: 'is_towable',
  scheduled_flag: 'is_scheduled',
  due_today_flag: 'is_due_today',
  vendor_personnel_full_name: 'vendor_name',
  issues: 'active_issue_flag',
};

export const getAvailableFilters = (bucket) => {
  const params = {
    filter_by: {
      dispatch_status: bucketToStatusMap[bucket],
    },
  };
  return tripsService
    .getFacetCount(params)
    .then(({ data }) => Promise.resolve(renameKeys(keymap, data)));
};

const prepareValues = (key, arr) => {
  const values = [];
  for (let i = 0; i < arr.length; i += 2) {
    values.push({
      code: arr[i],
      description: categoryMap[key].getKeys(arr[i]),
      count: arr[i + 1],
    });
  }
  return values;
};
export const prepareData = (data) => {
  const arr = [];
  forEachObjIndexed((val, key) => {
    if (categoryMap[key] && val.length > 0) {
      arr.push({
        field: categoryMap[key].filter || key,
        label: categoryMap[key].label,
        values: prepareValues(key, val) || val,
      });
    }
  }, data);
  return arr;
};

export const getSelectedCount = compose(reduce(add, 0), map(prop('count')), values);
