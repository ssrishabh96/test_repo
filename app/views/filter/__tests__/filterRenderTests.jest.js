import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Filter from '../filter.view';

import { getFacetCount } from 'views/trips/trips.service';
import * as helpers from '../filter.helpers';

jest.mock('views/trips/trips.service', () => ({
  getFacetCount: jest.fn(),
}));
getFacetCount.mockImplementation(() =>
  Promise.resolve({
    status: 'success',
    data: {
      damage_type_code: ['UK', 119, 'AO', 26, 'FR', 3],
      issues: ['No', 145, 'Yes', 3],
      location_city: [
        'LOVELAND',
        65,
        'DEER TRAIL',
        54,
        'DENVER',
        18,
        'WEATHERFORD',
        4,
        'GREENWOOD VILLAGE',
        2,
        'BELOIT',
        1,
        'FORT WORTH',
        1,
        'GRAND JUNCTION',
        1,
        'OKOLONA',
        1,
        'PUEBLO',
        1,
      ],
      payment_mode: ['CARD/CHECK', 117, 'CASH', 20, '', 11],
      towable_flag: ['Yes', 112, 'No', 34, 'Unknown', 2],
      trip_type_code: ['P', 144, 'D', 2, 'T', 2],
    },
  }),
);

const Props = {
  applyFilter: jest.fn(),
  bucket: 'bucket',
  selectedFilters: {},
  totalCount: 0,
  closeFilters: jest.fn(),
  isVisible: true,
  navigator: {
    showInAppNotification: jest.fn(),
  },
};
test('get facet count is only called when isVisible changes from false to true', () => {
  getFacetCount.mockImplementation(() =>
    Promise.resolve({
      status: 'success',
      data: {},
    }),
  );
  const component = shallow(<Filter {...{ ...Props, isVisible: false }} />);
  expect(getFacetCount.mock.calls.length).toEqual(0);
  component.setProps({ isVisible: true });
  expect(getFacetCount.mock.calls.length).toEqual(1);
  component.setProps({ isVisible: true });
  expect(getFacetCount.mock.calls.length).toEqual(1);
  component.setProps({ isVisible: false });
  expect(getFacetCount.mock.calls.length).toEqual(1);
});

test('the view renders correctly', () => {
  const tree = renderer.create(<Filter {...Props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('prepare data works', () => {
  const data = {
    active_issue_flag: ['No', 145, 'Yes', 3],
    trip_type_code: ['P', 144, 'D', 2, 'T', 2],
  };
  const expectedResult = [
    {
      field: 'active_issue_flag',
      label: 'Has Active Issue',
      values: [
        {
          code: 'No',
          count: 145,
          description: 'No',
        },
        {
          code: 'Yes',
          count: 3,
          description: 'Yes',
        },
      ],
    },
    {
      field: 'trip_type_code',
      label: 'Trip Type',
      values: [
        {
          code: 'P',
          count: 144,
          description: 'P',
        },
        {
          code: 'D',
          count: 2,
          description: 'D',
        },
        {
          code: 'T',
          count: 2,
          description: 'T',
        },
      ],
    },
  ];
  expect(helpers.prepareData(data)).toEqual(expectedResult);
});
