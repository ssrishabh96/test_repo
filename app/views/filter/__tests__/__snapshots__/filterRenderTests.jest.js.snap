// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the view renders correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(10, 8, 10, 0.78)",
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#005abc",
            "flexDirection": "row",
            "height": 65,
            "justifyContent": "space-between",
            "paddingHorizontal": 10,
            "paddingTop": 20,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
              }
            }
          >
            CLOSE
          </Text>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 17,
              "fontWeight": "bold",
            }
          }
        />
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
              }
            }
          >
            APPLY
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#323742",
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "paddingHorizontal": 5,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 16,
              "fontWeight": "bold",
              "padding": 2,
              "paddingBottom": 3,
            }
          }
        >
          0
           
          Selected
        </Text>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "rgb(248, 231, 28)",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "marginLeft": 10,
                }
              }
            >
              Reset
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <ActivityIndicator
          animating={true}
          color="#999999"
          hidesWhenStopped={true}
          size="large"
        />
      </View>
    </View>
  </View>
</Modal>
`;
